src/Microsoft.DotNet.TestFramework/Microsoft.DotNet.TestFramework.TestScenario.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.TestFramework
+{
+    public class TestScenario
```

Comment:
I recommend looking at this helper:

https://github.com/aspnet/dnx/blob/dev/test/Microsoft.Dnx.Testing.Framework/TestUtils.cs#L57

@JunTaoLuo spent a bunch of time making it so that we could write tests and have results stored per test in a usable fashion.